core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestSimplifyExpressions.java

```diff
@@ -69,6 +71,18 @@ public void testPushesDownNegations()
         assertSimplifies("NOT (X IS DISTINCT FROM Y)", "NOT (X IS DISTINCT FROM Y)");
     }
 
+    @Test
+    public void testSimplifyMatchAllWildCardLikePredicate()
```

Comment:
Can you please also provide tests for LIKE with constant  `name LIKE 'a constant'`?

Also can you provide tests with escape characters mixed with %?